src/Components/Blazor/Build/src/targets/StaticWebAssets.targets

```diff
@@ -0,0 +1,30 @@
+<Project>
+
+  <PropertyGroup>
+    <ResolveStaticWebAssetsInputsDependsOn>
+      $(ResolveStaticWebAssetsInputsDependsOn);
+      _RemoveBlazorWwwrootFromStaticWebAssets;
+    </ResolveStaticWebAssetsInputsDependsOn>
+  </PropertyGroup>
+
+
+  <Target Name="_RemoveBlazorWwwrootFromStaticWebAssets">
+    <ItemGroup>
+      <StaticWebAsset Remove="@(StaticWebAsset)" Condition="'%(SourceType)' == ''" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="BlazorStaticWebAssetsComputeFilesToPublish"
+    AfterTargets="_StaticWebAssetsComputeFilesToPublish">
+  
+    <ItemGroup>
+      <_StandaloneExternalPublishStaticWebAsset Include="@(_ExternalPublishStaticWebAsset)" Condition="'%(RelativePath)' != ''">
+        <RelativePath>$([MSBuild]::MakeRelative('$(MSBuildProjectDirectory)', '$([MSBuild]::NormalizePath('$([System.String]::Copy('%(RelativePath)').Replace('wwwroot','$(BlazorPublishDistDir)'))'))'))</RelativePath>
```

Comment:
The use of `Replace` here seems very dangerous. What if people have `wwwroot` in the path for some other reason too, e.g., they have `wwwroot/something/wwwroot/another`?

Reply:
I'm limited in the expressiveness here.

I would not consider wwwroot subpaths or subpaths that contain "wwwroot" as those are not common and it's not worth doing something more sophisticated.

If you feel very strongly about this, I can probably do this with a custom task, but I don't feel we need to go that far. If that were the case, I would simply suggest that we published everything to a wwwroot folder inside the publish folder as that would serve as a "subfolder" to isolate the output from the rest of the unwanted publish output and would be the folder we would tell people to upload to publish their app into Azure Blob Storage or similar.

The fact that today we use `$(TargetName)/dist` as a convention I feel complicates things a bit. If we end up not needing to have a separate folder, we will simply remove the /wwwroot from the relative path

Reply:
Would you mind adding a comment that says what this target is doing here with an example. e.g.

> Change any item that would have been published to wwwroot to appear in Blazor's publish output directory instead.

or something like that.

Reply:
I wouldn't block on this, but I wonder if there's a nicer way to do this.

Reply:
The paths are already made relative to `$(MSBuildProjectDirectory)`, and we only want to support `wwwroot` being directly inside that, so would it be possible to swap out the `Replace` with a regex match that only matches `wwwroot` at the start of the string, and where it's followed by a slash?

Reply:
@SteveSandersonMS I already replaced it with a regex but I still think this is overkill and that adds limited value. That said, it's done.